src/test/java/eu/rekawek/coffeegb/integration/support/ImageTestRunner.java

```diff
@@ -30,38 +35,101 @@ public class ImageTestRunner {
     public ImageTestRunner(File romFile) throws IOException {
         GameboyOptions options = new GameboyOptions(romFile, Collections.singletonList("grayscale"),Collections.emptyList());
         Cartridge cart = new Cartridge(options);
-        display = new SwingDisplay(1, options.isGrayscale());
+        display = new TestDisplay();
         gb = new Gameboy(options, cart, display, Controller.NULL_CONTROLLER, SoundOutput.NULL_OUTPUT, SerialEndpoint.NULL_ENDPOINT);
         imageFile = new File(romFile.getParentFile(),romFile.getName().replace(".gb",".png"));
     }
 
-    public boolean runTest() throws Exception {
-        new Thread(display).start();
+    public TestResult runTest() throws Exception {
         new Thread(gb).start();
         Thread.sleep(2000);
```

Comment:
In other test runners we're running gb.tick() in a loop, until some condition is true. Looking at the [dmg-acid2 source code](https://github.com/mattcurrie/dmg-acid2/blob/8a98ce731f96dde032ffb22ec36dc985d78fdb18/src/dmg-acid2.asm#L266-L268), it seems that such condition for this test is:

```
    ; source code breakpoint - good time to take a 
    ; screenshot to compare to reference image
    ld b, b
```

So we should run gb.tick(), until we get to the instruction `ld b,b`, which has opcode 0x40. The current opcode can be read with something like this:

```
int pc = gb.getCpu().getRegisters().getPc();
int opcode = gb.getAddressSpace().get(pc);
```

Doing so would allow us to remove the uncertainty coming from using threads and sleeps.

Reply:
A friend observing this PR suggested we should also add a watchdog that will fail the test after running certain number of cycles (to avoid hanging indefinitely if there's something wrong with emulator or testing ROM).